test.sh

```diff
@@ -146,7 +146,18 @@ GOBIN=${GOBIN:-$HOME/gopath/bin}
 #
 if [[ "$RUN" =~ "vet" ]] ; then
   start_context "test/vet"
-  run_and_comment go vet ./...
+  check_go_vet() {
```

Comment:
I think that's fine, you just adapt run_and_comment to return 1 if output is non-empty. That would also address JC's comment, below.